File: baseLib/src/main/java/me/ycdev/android/lib/common/perms/PermissionRequestParams.java
package me.ycdev.android.lib.common.perms;

import android.support.annotation.IntDef;
import android.support.annotation.StringRes;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class PermissionRequestParams {
    public static final int RATIONALE_POLICY_ON_DEMOND = 1;
    public static final int RATIONALE_POLICY_NEVER = 2;
    public static final int RATIONALE_POLICY_ALWAYS = 3;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({
        RATIONALE_POLICY_ON_DEMOND, RATIONALE_POLICY_NEVER, RATIONALE_POLICY_ALWAYS
    })
    public @interface RationalePolicy {}

    public int requestCode;
    public String[] permissions;
    public @RationalePolicy int rationalePolicy = RATIONALE_POLICY_ON_DEMOND;
    public String rationaleTitle;
    public String rationaleContent;
    public @StringRes int positiveBtnResId = android.R.string.ok;
    public @StringRes int negativeBtnResId = android.R.string.cancel;
    public PermissionCallback callback;
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/perms/PermissionRequestParams.kt
package me.ycdev.android.lib.common.perms

import androidx.annotation.IntDef
import androidx.annotation.StringRes

class PermissionRequestParams {

    var requestCode: Int = 0
    var permissions: Array<String>? = null
    @RationalePolicy
    var rationalePolicy = RATIONALE_POLICY_ON_DEMAND
    var rationaleTitle: String? = null
    var rationaleContent: String? = null
    @StringRes
    var positiveBtnResId = android.R.string.ok
    @StringRes
    var negativeBtnResId = android.R.string.cancel
    var callback: PermissionCallback? = null

    @Retention(AnnotationRetention.SOURCE)
    @IntDef(RATIONALE_POLICY_ON_DEMAND, RATIONALE_POLICY_NEVER, RATIONALE_POLICY_ALWAYS)
    annotation class RationalePolicy

    companion object {
        const val RATIONALE_POLICY_ON_DEMAND = 1
        const val RATIONALE_POLICY_NEVER = 2
        const val RATIONALE_POLICY_ALWAYS = 3
    }
}
